src/utils/form.ts

```diff
@@ -12,3 +16,129 @@ export const hiddenInputStyle = `
   -webkit-appearance: none !important;
   z-index: -1 !important;
 `;
+
+interface FormDataEvent extends Event {
+  formData: FormData;
+}
+
+/**
+ * Defines interface for components that are form-compatible.
+ *
+ * Along with the interface, use the matching form utils to help set up the component behavior.
+ */
+export interface FormAssociated<T = any> {
+  /**
+   * The host element.
+   */
+  readonly el: HTMLElement;
+
+  /**
+   * The form this component is associated with.
+   */
+  formEl: HTMLElement;
+
+  /**
+   * The name used to submit the value to the associated form.
+   *
+   * Note that this prop should use the @Prop decorator.
+   *
+   * @default ""
+   */
+  name: string;
+
+  /**
+   * The name of the property used to let the form know the component is toggled on.
+   */
+  onProperty?: string;
```

Comment:
@jcfranco I added this. Let me know your thoughts and if there's a better name.

Reply:
What's this for exactly?

Reply:
checked/switched

Reply:
Oh I see so that only exists because we use conflicting terminology.  Perhaps we should stop using 2 terms that mean the same thing and just use `checked` inside the switch component?

Reply:
That's an option but I think this is more flexible.